lighthouse-core/test/lib/sd-validation-test.js

```diff
@@ -169,7 +171,9 @@ describe('schema.org validation', () => {
     }`);
 
     assert.equal(errors.length, 1);
+    assert.equal(errors[0].invalidTypes[0], 'http://schema.org/Article');
```

Comment:
this name is a bit confusing to me for how it's used. the type isn't invalid, right? it's just the type of the entity that has some *other* invalid stuff going on?

Reply:
The entity is not a valid `type` instance because it has a property that doesn't exist on `type`.

I originally just called that array `types`, but then it wasn't clear that having that array means invalid. (And we use that array to generate the `Invalid Event: unexpected property asdf` message.)

Maybe we can call the array `invalidForTypes`?

Reply:
oh gotcha, I actually like the direction you were headed with `.types` then. I think it's clear *something* is invalid by virtue of the fact that we are giving an error. Maybe `typeOfInvalidEntity` or something similar if we're going the more explicit route?

Reply:
I'm at `validTypes` right now, since we only include the types in the validation error if they are valid. (And we don't want to show invalid types in the UI, because in that case they'll be part of the error message.)